src/Services/QueuedJobLogger.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Symbiote\QueuedJobs\Services;
+
+use Psr\Log\AbstractLogger;
+use Symbiote\QueuedJobs\DataObjects\QueuedJobDescriptor;
+
+/**
+ * Writes log output to a job descriptor
+ */
+class QueuedJobLogger extends AbstractLogger
+{
+    /** @var QueuedJob */
+    public $job;
+
+    /** @var QueuedJobDescriptor */
+    public $jobDescriptor;
+
+    public function __construct(QueuedJob $job, QueuedJobDescriptor $jobDescriptor)
+    {
+        $this->job = $job;
+        $this->jobDescriptor = $jobDescriptor;
+    }
+
+    /**
+     * Logs with an arbitrary level.
+     *
+     * @param mixed $level
+     * @param string $message
+     * @param array $context
+     *
+     * @return void
+     * @throws \SilverStripe\ORM\ValidationException
+     */
+    public function log($level, $message, array $context = array())
```

Comment:
```suggestion
    public function log($level, $message, array $context = [])
```

(stylistic suggestion)